### ISSUE TITLE
Package config type Number and minimum value doesn't always work

### ISSUE THREAD

**Issue by noseglid** [atom/atom#4990]
I've created a configuration variable in my package that looks like this:

``` javascript
    monocleHeight: {
      title: 'Monocle Height',
      description: 'How much of the workspace to use for build panel when it is "maximized".',
      type: 'number',
      default: 0.75,
      minimum: 0.1,
      maximum: 0.9
    }
```

Now, editing this variable in the Settings View, if I write `0`, it correctly changes it to `0.1`. However, if I write `0.01` it doesn't do .

Similar with maximum, if I write `1.4` it correctly changes it to `0.9`. Typing `2.0` and it doesn't change anything.

Good part: It never writes config with incorrect values, and never issues my config observe callbacks with incorrect values either. It's just bad feedback to the user.

**Comment by izuzak** [comment]
Thanks for taking the time to open this issue, @noseglid. I was looking into this today and wanted to leave some notes. @benogle @maxbrunsfeld I'm wondering if you have any thoughts about the problem and how to approach this, since I remember you walking these lands before.

I think this is a problem in how Settings View and the Config class communicate about changes. SettingsView uses [atom.config.observe](https://github.com/atom/settings-view/blob/5a115f9e051da32e7af676daf6076d890d8d5e75/lib/settings-panel.coffee#L102) to subscribe for config changes. The Config class listens on changes to config values internally and notifies subscribers, but [only if the new value is different from the old value](https://github.com/atom/atom/blob/e8b7a6bc1fb24b418e6e7d8f3666a9c5e35598e4/src/config.coffee#L892). 

This causes problems in cases where the observed config setting has an allowed range, and there's a UI observing the set value. The problem occurs when the current value of the config setting is already at the minimum or maximum, and the value that's being set is lower or higher than that, respectively. For example, imagine if the allowed maximum is `0.9` and you try to set the value to `2.0`. In that case, the config module notices that the new value is outside the allowed range, and [sets it to the edge of the range that is closest to the passed value](https://github.com/atom/atom/blob/e8b7a6bc1fb24b418e6e7d8f3666a9c5e35598e4/src/config.coffee#L1166-L1172). At that point, the new value becomes equal to the old value. Since there are no changes between the old value and the new value -- subscribers aren't notified, and the UI doesn't refresh itself.

In other words, the Settings View would also like to know if the new value of the config setting determined by the Config module is different from the value which was typed into the mini editor in Settings View, and not just if it's different from the old value of the config setting.

I don't consider this a huge priority, but it's definitely confusing in the UI. Also friendly cc @atom/feedback in case you have any ideas.
